### PR TITLE
Switch to ddebpf.Manager in consumer_test.go and related files

### PR DESCRIPTION
**Pull request by DanielLavie**
### What does this PR do?
Replaces manager.Manager with ddebpf.Manager in consumer_test.go and related files.

### Motivation
[This PR](https://github.com/DataDog/datadog-agent/pull/32358) is failing due to the missing usage of the ddebpf.Manager object in consumer_test.go.

### CODE REVIEW

File: pkg/network/protocols/events/consumer_test.go [hunk 1]
```diff
@@ -208,11 +209,13 @@ func newEBPFProgram(c *config.Config) (*manager.Manager, error) {
 		},
 	}
 
-	Configure(config.New(), "test", m, &options)
+	ddEbpfManager := ddebpf.NewManager(m, "usm", &ebpftelemetry.ErrorsTelemetryModifier{})
```
Comment (val06) [inline pkg/network/protocols/events/consumer_test.go:212]: do you really need the errorstelemetrymodifier for the test?

**Comment by usamasaqib** [comment]
Can you change the various functions like `NewConsumer` and `Configure` to work with `ddebpf.Manager` instead of passing `manager.Manager`? 
`ddebpf.Manager` imbeds the eBPF manager so you can directly use its methods. Moreover, for correct functioning it overloads some methods like `InitWithOptions`. Using `manager.Manager` may lead to bugs in the future where the incorrect methods get called.

In short, I would suggest to use `ddebpf.Manager` everywhere.

File: pkg/network/protocols/events/consumer_test.go (hunk 1 above)
Comment (usamasaqib) [reply pkg/network/protocols/events/consumer_test.go:212]: @val06  If the ebpf code uses the telemetry macros, the modifier is required to patch the code correctly.

**Review by usamasaqib — APPROVED** [review]

**Review by val06 — APPROVED** [review]
